icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/DecimalQuantityTest.java

```diff
@@ -603,8 +609,193 @@ public void testNickelRounding() {
         }
     }
 
+    @Test
+    public void testCompactDecimalSuppressedExponent() {
+        ULocale locale = new ULocale("fr-FR");
+        LocalizedNumberFormatter withLocale = NumberFormatter.withLocale(locale);
+        LocalizedNumberFormatter noCompactSimple =
+                withLocale.notation(Notation.simple());
+        LocalizedNumberFormatter compactLong =
+                withLocale.notation(Notation.compactLong());
+        LocalizedNumberFormatter compactShort =
+                withLocale.notation(Notation.compactShort());
+        LocalizedNumberFormatter scientific =
+                withLocale.notation(Notation.scientific());
+
+        Map<String,LocalizedNumberFormatter> formatters = new HashMap<>();
+        formatters.put("noCompactSimple", noCompactSimple);
```

Comment:
Suggestion: use skeletons instead of this hash map.  Makes it more flexible.

Reply:
Done.